engines/director/util.cpp

```diff
@@ -516,36 +517,57 @@ Common::String getPath(Common::String path, Common::String cwd) {
 	return cwd; // The path is not altered
 }
 
-bool testPath(Common::String &path, bool directory) {
-	Common::FSNode d = Common::FSNode(*g_director->getGameDataDir());
-	Common::FSNode node;
+Common::String convert83Path(Common::String &path) {
+	Common::String addedexts;
+	Common::String convPath;
 
-	// Test if we have it right in the SearchMan. Also accept MacBinary
-	// for Mac and Pippin
-	if (SearchMan.hasFile(Common::Path(path, g_director->_dirSeparator)) ||
-	    ((g_director->getPlatform() == Common::kPlatformMacintoshII
-	      || g_director->getPlatform() == Common::kPlatformMacintosh
-	      || g_director->getPlatform() == Common::kPlatformPippin) &&
-	     Common::MacResManager::exists(Common::Path(path, g_director->_dirSeparator))))
-		return true;
+	const char *ptr = path.c_str();
+	Common::String component;
 
-	debugN(9, "%s", recIndent());
-	debug(9, "testPath: %s  dir: %d", path.c_str(), directory);
+	while (*ptr) {
+		if (*ptr == g_director->_dirSeparator) {
+			if (component.equals(".")) {
+				convPath += component;
+			} else {
+				convPath += convertMacFilename(component.c_str());
+			}
 
-	// check for the game data dir
-	if (!path.contains(g_director->_dirSeparator) && path.equalsIgnoreCase(d.getName())) {
-		if (!directory)
-			return false;
-		path = "";
-		return true;
+			component.clear();
+			convPath += g_director->_dirSeparator;
+		} else {
+			component += *ptr;
+		}
+
+		ptr++;
 	}
 
+	if (hasExtension(component)) {
+		Common::String nameWithoutExt = component.substr(0, component.size() - 4);
+		Common::String ext = component.substr(component.size() - 4);
+		convPath += convertMacFilename(nameWithoutExt.c_str()) + ext;
+	} else {
+		convPath += convertMacFilename(component.c_str());
+	}
+
+	return convPath;
+}
+
+Common::FSNode resolvePath(Common::String &path, Common::FSNode &base, bool directory) {
+	Common::FSNode d;
+	if (base.exists() && base.isDirectory()) {
+		d = base;
+	} else {
+		d = Common::FSNode(*g_director->getGameDataDir());
```

Comment:
Making it work directly with FSNodes renders impossible use of SearchMan, with extra directories and makes mix-in of files via game quirks impossible.

Reply:
SearchMan and MacResManager have been added as fallbacks